File: Swift-C++-Testsuite/test/generics/Swift/InOutWithGenerics.swift
@main
struct MyMain {

    static func main() {
        var x = 42
        var y = 24
        swapTwoValues(&x, &y)
        // print("x: \(x) y: \(y)")
        var z = "World"
        var i = "Hello"
        swapTwoValues(&z, &i)
        // print(z + i)
    }

    static func swapTwoValues<T>(_ a: inout T, _ b: inout T) {
        let temporaryA = a
        a = b
        b = temporaryA
    }
}

File: AutoTranspiledTests/Swift-to-Cpp/try-0/InOutWithGenerics.cpp

#include <iostream>
#include <string>
    
template<typename T>
void swapTwoValues(T* a, T* b) {
    T tempA = *a;
    *a = *b;
    *b = tempA;
}

int main() {
    int x = 42;
    int y = 24;
    swapTwoValues(&x, &y);
    // std::cout << "x: " << x << " y: " << y << std::endl;
    std::string z = "World";
    std::string i = "Hello";
    swapTwoValues(&z, &i);
    // std::cout << z + i << std::endl;

    return 0;
}